### PR TITLE
Updated to Python 3 and Pycryptodome

### PR DESCRIPTION
**Pull request by bismurphy**
I updated all the code to Python 3 and also changed all the crypto routines to use pycryptodome instead of the deprecated pycrypto.

This is my first time doing anything on git, so if anything is wrong here, please let me know! I would love to have any of my procedural errors corrected. Thanks!

### CODE REVIEW

File: bin/psarc.py
Comment (0x0L) [inline bin/psarc.py]: Why not 'data + bytes(padding)'  ?

File: bin/psarc.py
Comment (0x0L) [inline bin/psarc.py]: This won't work

Comment (0x0L) [inline bin/psarc.py]: This need to be changed

File: bin/psarc.py
Comment (0x0L) [inline bin/psarc.py]: Same here

File: bin/psarc.py
Comment (0x0L) [inline bin/psarc.py]: `bytes(...)`

File: bin/psarc.py
Comment (0x0L) [inline bin/psarc.py]: `bytes()`

**Review by 0x0L — CHANGES_REQUESTED** [review]
Hi,

Thanks for your PR. I'm willing to merge this if you address the following issues:
1. fix the unecessary bytearray (we usually don't mutate anything so bytes is enough)
2. maintain python 2.7 compat

**Comment by tj-murphy** [comment]
Hi,

I went through and changed every bytearray to bytes. I also found a few chr(0)'s that I had missed earlier on, which have now been changed over. Not sure what your issue was with the spots where you said "This needs to be changed" or "same here" so I haven't made any changes yet.

Also not sure how I'm supposed to maintain Python 2.7 compatibility when the syntaxes are fundamentally different between the versions, particularly with how print statements work. If you could say where the issues are that would be great, or if it's not worth your time, feel free to just say "I don't want this PR".

Thanks!

**Comment by 0x0L** [comment]
There are several ways to maintain compatibility with python 2.7 (see http://python-future.org/compatible_idioms.html). For the print function it's really easy. The other possible source of problem here would be the confusion between bytes and str.

**Comment by tj-murphy** [comment]
After a bit of research, I've gotten things running in both Python 2 and Python 3 environments. I also removed all instances of bytearray. I believe that handles the issues raised?

Please let me know if you'd like any further changes made.

**Comment by 0x0L** [comment]
Thanks, looks good.
Could you please squash all the commits before I merge ? Thanks

**Comment by tj-murphy** [comment]
Okay, I think I squashed commits. This is the first time I've done that so may have done something wrong but I think it worked; please let me know if not. Thanks!

**Review by 0x0L — APPROVED** [review]

**Comment by 0x0L** [comment]
Thanks for the PR

**Pull request merged**
